src/content/symbol-store-db.js

```diff
@@ -1,134 +1,250 @@
-      const transaction = this._db.transaction('symbol-tables', 'readonly');
+      const transaction = db.transaction('symbol-tables', 'readwrite');
       const store = transaction.objectStore('symbol-tables');
-      const index = store.index('libKey');
-      const req = index.openKeyCursor(IDBKeyRange.only([debugName, breakpadId]));
-      req.onerror = reject;
-      req.onsuccess = () => {
-        const cursor = req.result;
+      this._findRecordsLastUsedBeforeDate(store, beforeDate, primaryKeysToDelete => {
+        this._deleteRecords(store, primaryKeysToDelete, () => {
+          resolve();
+        }).catch(reject);
+      }).catch(reject);
+    });
+  }
+
+  _findRecordsLastUsedBeforeDate(store: IDBObjectStore, beforeDate: Date, callback: (number[]) => void): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const lastUsedDateIndex = store.index('lastUsedDate');
+      // Get a cursor that walks all records whose lastUsedDate is less than
+      // beforeDate.
+      const cursorReq = lastUsedDateIndex.openKeyCursor(
+        window.IDBKeyRange.upperBound(beforeDate, true));
+      // Iterate over all records in this cursor and collect their primary
+      // keys into allFoundKeys.
+      const allFoundKeys = [];
+      cursorReq.onsuccess = () => {
+        const cursor: IDBCursor = (cursorReq: any).result;
         if (cursor) {
-          resolve(cursor.primaryKey);
-        } else {
-          reject(new Error('Nothing in the async cache for this library.'));
+          allFoundKeys.push(cursor.primaryKey);
+          cursor.continue();
+          return;
         }
+        callback(allFoundKeys);
+        resolve();
+      };
+      cursorReq.onerror = () => {
+        reject(cursorReq.error);
       };
     });
   }
 
-  importLibrary(debugName, breakpadId, [addrs, index, buffer]) {
-    if (!this._db) {
-      return this._setupDBPromise.then(() => this.importLibrary(debugName, breakpadId, [addrs, index, buffer]));
-    }
-
+  _findNLeastRecentlyUsedRecords(store: IDBObjectStore, n: number, callback: (number[]) => void): Promise<void> {
     return new Promise((resolve, reject) => {
-      const transaction = this._db.transaction('symbol-tables', 'readwrite');
-      transaction.onerror = reject;
-      const tableStore = transaction.objectStore('symbol-tables');
-      const putReq = tableStore.put({ debugName, breakpadId, addrs, index, buffer });
-      putReq.onsuccess = () => {
-        resolve(putReq.result);
+      const lastUsedDateIndex = store.index('lastUsedDate');
+      // Get a cursor that walks the records from oldest to newest
+      // lastUsedDate.
+      const cursorReq = lastUsedDateIndex.openKeyCursor();
+      const allFoundKeys = [];
+      cursorReq.onsuccess = () => {
+        const cursor: IDBCursor = (cursorReq: any).result;
+        if (cursor) {
+          allFoundKeys.push(cursor.primaryKey);
+          if (allFoundKeys.length < n) {
```

Comment:
ah right !

that's a quite complex machinery for a difference of only 2 lines. I'll let you decide if it's worth generalizing this as you know this code a lot more :)